hub-detect/src/test/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnListParserTest.java

```diff
@@ -87,12 +89,15 @@ public void testDependencyInYarnListAndNotInLock() {
         testLines.add("yarn list v1.5.1");
         testLines.add("├─ abab@1.0.4");
 
-        final YarnListParser yarnListParser = new YarnListParser();
+        final ExternalIdFactory externalIdFactory = new ExternalIdFactory();
+        final YarnLockParser yarnLockParser = new YarnLockParser();
+        final YarnListParser yarnListParser = new YarnListParser(externalIdFactory, yarnLockParser);
         final DependencyGraph dependencyGraph = yarnListParser.parseYarnList(designedYarnLock, testLines);
 
         final List<ExternalId> tempList = new ArrayList<>(dependencyGraph.getRootDependencyExternalIds());
 
-        assertEquals(0, tempList.size());
+        assertEquals(1, tempList.size());
```

Comment:
Did we change the meaning of this test?

Reply:
Not as such, instead Yarn's logic changed. A majority of these tests were verifying the old (and incorrect) behavior of Yarn.